src/commands/gen-api-models/parse.ts

```diff
@@ -3,17 +3,119 @@
  */
 
 import { ITuple2, Tuple2, Tuple3 } from "@pagopa/ts-commons/lib/tuples";
-import { OpenAPIV2 } from "openapi-types";
+import { OpenAPIV2, IJsonSchema } from "openapi-types";
 import { uncapitalize } from "../../lib/utils";
 import {
   ExtendedOpenAPIV2SecuritySchemeApiKey,
   IAuthHeaderParameterInfo,
+  IDefinition,
   IHeaderParameterInfo,
   IOperationInfo,
   IParameterInfo,
   ISpecMetaInfo,
   SupportedMethod
 } from "./types";
+
+// eslint-disable-next-line prefer-arrow/prefer-arrow-functions
+function parseInnerDefinition(source: IJsonSchema): IDefinition {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const looselySource = source as any;
+
+  if (looselySource.$ref) {
+    return looselySource;
+  }
+
+  // OAS2 does not support disjointed unions with oneOf,
+  //  so we introduced "x-one-of" custom field in association with allOf
+  const { oneOf, allOf } =
+    typeof looselySource.oneOf === "undefined" &&
+    "x-one-of" in looselySource &&
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    looselySource["x-one-of"]
+      ? { allOf: undefined, oneOf: looselySource.allOf }
```

Comment:
Not sure about it, but i don't think `oneOf: looselySource.allOf` is right

```suggestion
      ? { allOf: undefined, oneOf: looselySource["x-one-of"] }
```

Reply:
Nope, it's not the semantic of `x-one-of`. It's a boolean meant to tell the provided `allOf`must be actually handled as a `oneOf`

Reply:
Oh right, I was forgetting that this was our way to implement the `oneOf` in V2, my bad.